tests/nightly/test_all.sh

```diff
@@ -72,9 +80,15 @@ check_val() {
 example_dir=../../example/image-classification
 # python: lenet + mnist
 test_lenet() {
-    python $example_dir/train_mnist.py \
-        --data-dir `pwd`/data/mnist/ --network lenet --gpus $gpus --num-epochs 10 \
-        2>&1 | tee log
+    optimizer=$(get_random_optimizer)
```

Comment:
You mean back-to-back or as part of the stochastic set?  Back-to-back would be too slow. 

The individual optimizers themselves are tested as part of the unit tests in test_optimizer.py (which doesn't use fit.py logic).

I selected these three because they're the most common and are mostly just testing the fit.py logic. They all have much different parameters from each other, so should catch problems with parameters (which is what caused this problem).  My logic was by adding too many optimizers, the listed optimizers could go a very long time (more than a week) without being tested, and failures would seem much more random.  For example, SGD is the default and several optimizers are derived from SGD.

It's certainly possible to test them all, however.